PlatformIO and some refactors

jvegaf opened the pull request:
Hi, 

 I've migrated the Arduino project to PlatformIO because its more feats like:

- deps management more easy, with automatic downloads and updates
- no plugin needed to upload the data folder ( we have a task ), it's explained in the README.
- have a power linter tool that helps to avoid have bugs in the code....

![image](https://user-images.githubusercontent.com/1075386/161354402-fd520275-0d26-4dc3-a7ca-0d8dce91f527.png)


---

test with my nodemcu board is working perfect 😄 
![image](https://user-images.githubusercontent.com/1075386/161353889-9650a7a7-0bff-472f-b12e-168497ce9329.png)


Coming soon will like send a some PR with more little refactors

Regards

jvegaf committed:
added git ignore file

jvegaf committed:
first refactor

jvegaf committed:
update readme

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
refactor(extract includes and replace ifdef by pragma ):

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
extract to helper file

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
remove death code

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
remove old wifi settings and method

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
move example to doc folder

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
Revert "move example to doc folder"

This reverts commit 8f58dc12ccac79f0982fadadea4715b3d1c69bed.

jvegaf committed:
moved example to doc directory

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
push defines to config file

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
ntptimer decoupling

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
extract to wifi module

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
OTA module

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
decoupling sensor devices

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
hotfix

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
reduce float decimal places

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf committed:
hotfix

Signed-off-by: Jose Vega <josevega234@gmail.com>

jvegaf commented:
Sorry, I forget extract in other branch the code refactors, even so I did testing and commit each little change for avoid broke nothing. 

The main reason of this refactors is separate responsibilities in different modules, making this great project more testable and robust, so that will be more easy add new features.

issalig commented:
Hi, thanks for your work. It is nice to see other people collaborating.
I'm busy these days but I will take a look as soon as possible. 
Is it possible to have the code for Arduino IDE and PlatformIO? Last time I tried PlatformIO I had to change .ino extensions and some other things but I would like to have it working for both frameworks.

jvegaf commented:
> 
I'll look at a way to make it compatible with both environments.

in my case when l tried platformIO didn't feel like back to arduino.

jvegaf commented:
I found the solution 🚀🚀🚀 

now works with both environments.

compiled with Arduino 👇🏻 

![image](https://user-images.githubusercontent.com/1075386/162075460-f19e1821-73e6-4074-aa1d-ecfba30b55a8.png)

jvegaf committed:
Feat: dual compatibilities

Hack for can works with Arduino IDE and PlatformIO at the same time

Also documented

Signed-off-by: Jose Vega <josevega234@gmail.com>

issalig commented:
Great. I will take a look asap

issalig commented:
Sorry, still no time.

jvegaf commented:
> Sorry, still no time.

Don't worry friend, that's things happen to everybody. Trust me, I prefer that you take it with calm because you better understand the changes that I propose you.  

Now I am doing a few changes to the PCB for achieve works with a NodeMCU board (I have many of this boards forgotten in a drawer 😄 ). 

I will make a pcb piece when I end the changes to continue playing with the project.

snetekombinerke commented:
Hi, Is is possible to get finished pcb?

issalig commented:
Hi @snetekombinerke , if you are interested in my version for wemos d1 mini write me for more details about getting the pcb. If it is the nodeMCU, just contact jvegaf.

jvegaf commented:
> Hi @snetekombinerke , if you are interested in my version for wemos d1 mini write me for more details about getting the pcb. If it is the nodeMCU, just contact jvegaf.

@issalig  @snetekombinerke  just I released the esp32 pcb version 😁 

[https://github.com/jvegaf/toshiba_air_cond/releases/tag/esp32_v1](https://github.com/jvegaf/toshiba_air_cond/releases/tag/esp32_v1)

![airpcb_smd_board](https://user-images.githubusercontent.com/1075386/186792459-749bcc4e-2cdc-4e97-9b6b-46c6bad41425.png)


![airpcb_smd_top](https://user-images.githubusercontent.com/1075386/186792484-675153ea-c572-44ec-b5aa-e914440b4819.png)


![airpcb_smd_bottom](https://user-images.githubusercontent.com/1075386/186792492-6c0baa0d-4463-4344-aeb0-b9833b292bdb.png)

issalig commented:
Nice!

@jvegaf, BTW I wrote you an email to your gmail some weeks ago, not sure if you got it.

jvegaf commented:
> Nice!
> 
> @jvegaf, BTW I wrote you an email to your gmail some weeks ago, not sure if you got it. 

I've searched in my gmail but I doesn't found it.

@issalig  can you resend me it again ?

Thanks

issalig commented:
Is josevega234  correct? it is where I sent it.

jvegaf commented:
I sended you a email to your mejijof at getnada dot com account

issalig commented:
Ok. Got it. 
It seems I am writing to the correct address, anyway
You can also write me to issalig AT yahoo DOT es

jvegaf commented:
I replied your mail 😊

snetekombinerke commented:
Hi, I've sent you an email 30. avg., 21:39. Please check.

jvegaf commented:
Yes gmail dot com

El sáb, 27 ago 2022, 12:51, issalig ***@***.***> escribió:

> Is josevega234 correct? it is where I sent it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/issalig/toshiba_air_cond/pull/12#issuecomment-1229170040>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIGROWBWW636QAC4LBCA33V3HXKBANCNFSM5SKBCWYA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

obvious-creations commented:
Hi @jvegaf, was just wondering how far did you get this? I was looking out for solutions for our Toshiba air conditioner units and stumbled across this project

issalig commented:
Closed this as the new version also supports Platformio.
